docs/source/literal_types.rst

```diff
@@ -289,6 +289,74 @@ using ``isinstance()``:
 This feature is sometimes called "sum types" or "discriminated union types"
 in other programming languages.
 
+Exhaustive checks
+*****************
+
+One may want to check that some code covers all possible ``Literal`` cases, example:
+
+.. code-block:: python
+
+  from typing import Literal
+
+  PossibleValues = Literal['one', 'two']
+
+  def validate(x: PossibleValues) -> bool:
+      if x == 'one':
+          return True
+      elif x == 'two':
+          return False
+      raise ValueError('Wrong values passed: {0}'.format(x))
+
+  assert validate('one') is True
+  assert validate('two') is False
+
+In the code above it is really easy to make a mistake in the future: 
+by adding a new literal value to ``PossibleValues``, 
+but not adding its handler to ``validate`` function:
+
+.. code-block:: python
+
+  PossibleValues = Literal['one', 'two', 'three']
+
+Mypy won't catch that ``'three'`` is not covered.
+However, if you want to have exhaustive check, you need to guard it properly:
+
+.. code-block:: python
+
+  from typing import Literal, NoReturn
+
+  PossibleValues = Literal['one', 'two']
+
+  def assert_exhaustive(value: NoReturn) -> NoReturn:
```

Comment:
People actually often call this function `assert_never()` it can be used is some other cases where people want to check statically some will not be executed.